src/Materials/Node/Blocks/trigonometryBlock.ts

```diff
@@ -65,8 +65,8 @@ export class TrigonometryBlock extends NodeMaterialBlock {
     public constructor(name: string) {
         super(name, NodeMaterialBlockTargets.Neutral);
 
-        this.registerInput("input", NodeMaterialBlockConnectionPointTypes.Float);
-        this.registerOutput("output", NodeMaterialBlockConnectionPointTypes.Float);
+        this.registerInput("input", NodeMaterialBlockConnectionPointTypes.AutoDetect);
+        this.registerOutput("output", NodeMaterialBlockConnectionPointTypes.BasedOnInput);

```

Comment:
haha you need to connect the output with input still

Reply:
can you give me a quick example?

Sorry I am trying to do it from just using the other blocks as an example.

Is this what you are taking about:
this._outputs[0]._typeConnectionSource = this._inputs[0];